### ISSUE TITLE
Correct way to pass user data to AWS AMI instance

### ISSUE THREAD

**Issue by shivarammysore** [coreos/fedora-coreos-tracker#428]
Can you please point me to a document where I can make sure I am doing the right way in passing ignition file to AWS AMI instance?

I have a `t2.micro` instance of https://console.aws.amazon.com/ec2/home?region=us-east-1#launchAmi=ami-0c95394f5bdc9e749 
I want to pass a ignition file https://gist.github.com/shivarammysore/37f90a401b3a3236c2f4c2515aa04821 and I select an existing AWS key to login.  If I have any of this user data, my instance does not allow me to login.  The moment I remove the user data, all is well - of course, I have to manually change the files to get my settings.

**Comment by bgilbert** [comment]
It sounds as though Ignition is unable to apply your config.  Try stopping the instance and then fetching its console log to see the error.

**Comment by dustymabe** [comment]
random thought: t2.micro only has 1G of memory which might be causing inconsistency for you (I think I saw someone else trying the same thing recently and memory was a factor). Just to rule that out could you try on say `t2.medium` just to see.

**Comment by shivarammysore** [comment]
We have been running CoreOS (pre RedHat) for 3+ years with t2.nano and
t2.micro without issues.  If we need to start using t2.medium for passing
ignition files, we really have to look at alternatives.  It simply won't
fit our cost profiles :-(

**Comment by dustymabe** [comment]
@shivarammysore - I don't know if we given a lot of thought to an actual minimum required amount of RAM for an instance of FCOS but we should probably work on some guidelines there.

Regardless, you giving `t2.medium` a try and see if that works for you will help us debug the current issue you are seeing.

**Comment by dustymabe** [comment]
ahh it turns out it was you who who was trying this with `t2.micro`: https://github.com/coreos/fedora-coreos-tracker/issues/423#issuecomment-599573280

and @jlebon seemed to confirm he could boot with `t2.micro` so that is most likely not the problem.

**Comment by dustymabe** [comment]
ok I tried with your config locally. Here's what I see:

```
CRITICAL : Ignition failed: failed to create files: failed to create files: error creating /sysroot/etc/chrony.conf: error creating file "/sysroot/etc/chrony.conf": A file exists there already and overwrite is false
```

The Ignition spec v3 changed the default overwrite behavior to `false` ([docs link](https://github.com/coreos/ignition/blob/master/doc/migrating-configs.md#files-now-default-to-overwritefalse))


It works if I modify the ingition file like:

```diff
--- awsfcos.ign.orig    2020-03-19 09:34:46.296706092 -0400
+++ awsfcos.ign 2020-03-19 09:35:15.539046509 -0400
@@ -28,6 +28,7 @@
   "storage": {
     "files": [
       {
+        "overwrite": true,
         "group": {},
         "path": "/etc/selinux/config",
         "user": {},
@@ -38,6 +39,7 @@
         "mode": 272
       },
       {
+        "overwrite": true,
         "group": {},
         "path": "/etc/chrony.conf",
         "user": {},
```

**Comment by shivarammysore** [comment]
@dustymabe thanks for the help and clarification.  I tested the same and it works.  

Below is how I modified the yaml file.  I first did the normal YAML thing (without overwrite and source inline) and generated the generated the ignition file.  Saved this file.  This will enable me to use the URL encoded escaped sequence that I can cut-n-paste into the next step.  Then, I modified the YAML to look like this:

```yaml
storage:
  files:
    - path: /etc/selinux/config
      overwrite: true
      contents: 
        source: data:text/plain;charset=iso-8859-7,<paste_the_source_content_from_saved_ignition_file>
      mode: 0420
```

Now generate the ignition file with `fcct` again and use this file in `user data` to launch FCOS instance.

Suggestion: possibly add this as example to https://github.com/coreos/fcct/blob/master/docs/examples.md

Thanks again for the help.
